osquery/tables/system/windows/userassist.cpp

```diff
@@ -0,0 +1,179 @@
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/tables/system/windows/userassist.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/system/time.h>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+constexpr auto kFullRegPath =
+    "\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\UserAssist";
+
+// Decode ROT13 sub key value
+std::string rot_decode(std::string& value_key_reg) {
+  std::string decoded_value_key;
+
+  for (std::size_t i = 0; i < value_key_reg.size(); i++) {
+    if (isalpha(value_key_reg[i])) {
+      if (value_key_reg[i] >= 'a' && value_key_reg[i] <= 'm') {
+        decoded_value_key.append(1, value_key_reg[i] + 13);
+      } else if (value_key_reg[i] >= 'm' && value_key_reg[i] <= 'z') {
+        decoded_value_key.append(1, value_key_reg[i] - 13);
+      } else if (value_key_reg[i] >= 'A' && value_key_reg[i] <= 'M') {
+        decoded_value_key.append(1, value_key_reg[i] + 13);
+      } else if (value_key_reg[i] >= 'M' && value_key_reg[i] <= 'Z') {
+        decoded_value_key.append(1, value_key_reg[i] - 13);
+      }
+    } else {
+      decoded_value_key.append(1, value_key_reg[i]);
+    }
+  }
+  return decoded_value_key;
+}
+
+// Get last exeution time
+auto last_execute(std::string& assist_data) {
+  std::string last_run_string = assist_data.substr(120, 16);
+
+  // If timestamp is zero dont convert to UNIX Time
+  if (last_run_string == "0000000000000000") {
+    std::string no_timestamp = "";
+    return no_timestamp;
+  } else {
+    // swap endianess
+    std::reverse(last_run_string.begin(), last_run_string.end());
+
+    char temp;
+    for (std::size_t i = 0; i < last_run_string.length(); i += 2) {
+      temp = last_run_string[i];
+      last_run_string[i] = last_run_string[i + 1];
+      last_run_string[i + 1] = temp;
+    }
+
+    // Convert Windows FILETIME to UNIX Time
+    unsigned long long last_run = std::stoull(last_run_string.c_str(), 0, 16);
+    last_run = (last_run / 10000000) - 11644473600;
+
+    std::time_t last_run_time = last_run;
+
+    struct tm tm;
+    gmtime_s(&tm, &last_run_time);
+
+    auto time_str = platformAsctime(&tm);
+    return time_str;
+  }
+}
+
+// Get execution count
+int execution_num(std::string& execution_count) {
+  // swap endianess
+  std::reverse(execution_count.begin(), execution_count.end());
+
+  char temp;
+  for (std::size_t i = 0; i < execution_count.length(); i += 2) {
+    temp = execution_count[i];
+    execution_count[i] = execution_count[i + 1];
+    execution_count[i + 1] = temp;
+  }
+
+  auto count = tryTo<std::size_t>(execution_count, 16);
+  if (count.isError()) {
+    LOG(WARNING) << "Error getting execution count: " << count.takeError();
+    return -1;
+  }
+  return count.get();
+}
+
+QueryData genUserAssist(QueryContext& context) {
+  QueryData results;
+  QueryData users;
+
+  queryKey("HKEY_USERS", users);
```

Comment:
Is the intention to search for each user on the system?

Reply:
yes